node-repository/src/test/java/com/yahoo/vespa/hosted/provision/provisioning/ProvisioningTester.java

```diff
@@ -393,12 +382,75 @@ private Flavor getNodeFlavor(String hostname) {
         return nodeRepository.getNode(hostname).map(Node::flavor).orElseThrow(() -> new RuntimeException("No flavor for host " + hostname));
     }
 
-    private static class NullProvisionLogger implements ProvisionLogger {
+    public static final class Builder {
```

Comment:
The idea is to have a builder where you can set certain components that are relevant to your test, and other components will be set to some reasonable default. This is particularly useful when adding a component that you want to test somewhere, then this builder can set some useful defaults that let all the existing tests work without changes, while the new test could set some custom value.

Maybe overtime we could then merge the many `allocate()` methods we have spread out over the different tests and wont have to change bunch of tests every time we add a component to `NodeRepositoryProvisioner`...